DeserializerException: Error parsing response body

I'm just calling the TestDriver.main() method and I'm getting a DeserializerException.
I have 2 Repeaters and 1 PowerLine connected to my network. My router is on the newest version.
Sometimes I get a SocketTimeoutException instead.

Full log:
```
> Task :Main.main()
17:02:18.534 [main] DEBUG com.github.kaklakariada.fritzbox.TestDriver - Reading config from file H:\Users\Devin\Desktop\Fritz\application.properties
17:02:18.539 [main] INFO com.github.kaklakariada.fritzbox.TestDriver - Logging in to 'https://fritz.box' with username 'fritz4764'
17:02:20.228 [main] DEBUG com.github.kaklakariada.fritzbox.FritzBoxSession - Got response 045e4ad8-01d69cc698b1bbd7335c91b806ffb51c for challenge 045e4ad8
17:02:20.612 [main] DEBUG com.github.kaklakariada.fritzbox.FritzBoxSession - Logged in with session id 0622a1303fe229f7 and rights [UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0], UserRight [name=null, access=0]]
Exception in thread "main" com.github.kaklakariada.fritzbox.mapping.DeserializerException: Error parsing response body '<devicelist version="1" fwversion="7.29"></devicelist>'
	at com.github.kaklakariada.fritzbox.mapping.Deserializer.parse(Deserializer.java:59)
	at com.github.kaklakariada.fritzbox.http.HttpTemplate.parse(HttpTemplate.java:104)
	at com.github.kaklakariada.fritzbox.http.HttpTemplate.get(HttpTemplate.java:82)
	at com.github.kaklakariada.fritzbox.http.HttpTemplate.get(HttpTemplate.java:71)
	at com.github.kaklakariada.fritzbox.FritzBoxSession.getAutenticated(FritzBoxSession.java:90)
	at com.github.kaklakariada.fritzbox.HomeAutomation.executeCommand(HomeAutomation.java:57)
	at com.github.kaklakariada.fritzbox.HomeAutomation.getDeviceListInfos(HomeAutomation.java:49)
	at com.github.kaklakariada.fritzbox.TestDriver.main(TestDriver.java:51)
	at Main.main(Main.java:6)
Caused by: org.simpleframework.xml.core.ValueRequiredException: Unable to satisfy @org.simpleframework.xml.ElementList(entry="", data=false, inline=true, name="device", type=com.github.kaklakariada.fritzbox.model.homeautomation.Device.class, required=true, empty=true) on field 'devices' private java.util.List com.github.kaklakariada.fritzbox.model.homeautomation.DeviceList.devices for class com.github.kaklakariada.fritzbox.model.homeautomation.DeviceList at line 1
	at org.simpleframework.xml.core.Composite.validate(Composite.java:644)
	at org.simpleframework.xml.core.Composite.readElements(Composite.java:449)
	at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
	at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
	at org.simpleframework.xml.core.Composite.read(Composite.java:201)
	at org.simpleframework.xml.core.Composite.read(Composite.java:148)
	at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
	at org.simpleframework.xml.core.Persister.read(Persister.java:625)
	at org.simpleframework.xml.core.Persister.read(Persister.java:606)
	at org.simpleframework.xml.core.Persister.read(Persister.java:584)
	at org.simpleframework.xml.core.Persister.read(Persister.java:562)
	at org.simpleframework.xml.core.Persister.read(Persister.java:499)
	at org.simpleframework.xml.core.Persister.read(Persister.java:408)
	at com.github.kaklakariada.fritzbox.mapping.Deserializer.parse(Deserializer.java:55)
	... 8 more

Caused by: org.simpleframework.xml.core.ValueRequiredException: Unable to satisfy @org.simpleframework.xml.ElementList(entry="", data=false, inline=true, name="device", type=com.github.kaklakariada.fritzbox.model.homeautomation.Device.class, required=true, empty=true) on field 'devices' private java.util.List com.github.kaklakariada.fritzbox.model.homeautomation.DeviceList.devices for class com.github.kaklakariada.fritzbox.model.homeautomation.DeviceList at line 1

> Task :Main.main() FAILED

Execution failed for task ':Main.main()'.
> Process 'command 'D:/Program Files/Java/jdk-14.0.1/bin/java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights. 
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * [#29](https://github.com/kaklakariada/fritzbox-java-api/issues/29) Add support for login via Pbkdf2 challenge-response (propsed by [@linnex81](https://github.com/linnex81))
 
+### Fixed
+
+* [#37](https://github.com/kaklakariada/fritzbox-java-api/issues/37) Support empty device lists
+
 ### Refactoring
 
 * [#50](https://github.com/kaklakariada/fritzbox-java-api/pull/50) Migrate to Junit 5

--- a/src/main/java/com/github/kaklakariada/fritzbox/model/homeautomation/DeviceList.java
+++ b/src/main/java/com/github/kaklakariada/fritzbox/model/homeautomation/DeviceList.java
@@ -19,6 +19,7 @@ package com.github.kaklakariada.fritzbox.model.homeautomation;
 
 import static java.util.stream.Collectors.toList;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import org.simpleframework.xml.Attribute;
@@ -31,8 +32,8 @@ public class DeviceList {
     @Attribute(name = "version")
     private String apiVersion;
 
-    @ElementList(name = "device", type = Device.class, inline = true)
-    private List<Device> devices;
+    @ElementList(name = "device", type = Device.class, inline = true, required = false)
+    private List<Device> devices = new ArrayList<>();
 
     @ElementList(name = "group", type = Group.class, inline = true, required = false)
     private List<Group> groups;

--- a/src/test/java/com/github/kaklakariada/fritzbox/mapping/DeserializerTest.java
+++ b/src/test/java/com/github/kaklakariada/fritzbox/mapping/DeserializerTest.java
@@ -91,6 +91,12 @@ class DeserializerTest {
     }
 
     @Test
+    void parseDeviceListEmpty() throws IOException {
+        final DeviceList deviceList = parseDeviceList(Paths.get("src/test/resources/FritzOS29/deviceListEmpty.xml"));
+        assertThat(deviceList.getDevices()).hasSize(0);
+    }
+
+    @Test
     void parseDeviceList() throws IOException {
         final DeviceList deviceList = parseDeviceList(Paths.get("src/test/resources/deviceList.xml"));
         assertThat(deviceList.getDevices()).hasSize(16);

--- a/src/test/resources/FritzOS29/deviceListEmpty.xml
+++ b/src/test/resources/FritzOS29/deviceListEmpty.xml
@@ -1,0 +1,1 @@
+<devicelist version="1" fwversion="7.29"></devicelist>

